cli/replicasets/setup.go

```diff
@@ -0,0 +1,304 @@
+			*editReplicasetsOpts = append(*editReplicasetsOpts, editReplicasetOpts)
+		} else {
+			editReplicasetOpts, err := getUpdateReplicasetEditReplicasetsOpts(topologyReplicaset, replicasetConf, instancesConf)
+			if err != nil {
+				return nil, fmt.Errorf("Failed to get edit_topology options for updating replicaset: %s", err)
+			}
+			*editReplicasetsOpts = append(*editReplicasetsOpts, editReplicasetOpts)
+		}
+	}
+
+	newTopologyReplicasets, err := editReplicasets(conn, editReplicasetsOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	return newTopologyReplicasets, nil
+}
+
+func setFailoverPriority(conn net.Conn, replicasetsList *ReplicasetsList, topologyReplicasets *TopologyReplicasets) (*TopologyReplicasets, error) {
+	editReplicasetsOpts := EditReplicasetsOpts{}
+
+	for _, replicasetConf := range *replicasetsList {
+		newTopologyReplicaset := topologyReplicasets.GetByAlias(replicasetConf.Alias)
+
+		// set failover priority
+		editReplicasetOpts, err := getSetFailoverPriorityEditReplicasetOpts(replicasetConf.InstanceNames, newTopologyReplicaset)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to get edit_topology options for setting failover priority: %s", err)
+		}
+
+		editReplicasetsOpts = append(editReplicasetsOpts, editReplicasetOpts)
+	}
+
+	newTopologyReplicasets, err := editReplicasets(conn, &editReplicasetsOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	return newTopologyReplicasets, nil
+}
+
+func logSetupSummary(topologyReplicasets, newTopologyReplicasets *TopologyReplicasets) {
+	for replicasetUUID, newTopologyReplicaset := range *newTopologyReplicasets {
+		replicasetRes := common.Result{
+			ID: newTopologyReplicaset.Alias,
+		}
+
+		if oldTopologyReplicaset, found := (*topologyReplicasets)[replicasetUUID]; !found {
+			replicasetRes.Status = common.ResStatusCreated
+		} else if reflect.DeepEqual(newTopologyReplicaset, oldTopologyReplicaset) {
+			replicasetRes.Status = common.ResStatusOk
+		} else {
+			replicasetRes.Status = common.ResStatusUpdated
+		}
+
+		log.Infof(replicasetRes.String())
+	}
+}
+
+func getReplicasetsList(ctx *context.Ctx) (*ReplicasetsList, error) {
+	var err error
+
+	if _, err := os.Stat(ctx.Replicasets.File); err != nil {
+		return nil, fmt.Errorf("Failed to use replicasets configuration file: %s", err)
+	}
+
+	fileContentBytes, err := common.GetFileContentBytes(ctx.Replicasets.File)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read replicasets configuration file: %s", err)
+	}
+
+	var replicasetsConf ReplicasetsConf
+	if err := yaml.Unmarshal([]byte(fileContentBytes), &replicasetsConf); err != nil {
+		return nil, fmt.Errorf("Failed to parse replicasets configuration file %s: %s", ctx.Replicasets.File, err)
+	}
+
+	replicasetsList := make(ReplicasetsList, len(replicasetsConf))
+
+	i := 0
+	for replicasetAlias, replicasetConf := range replicasetsConf {
+		replicasetConf.Alias = replicasetAlias
+
+		replicasetsList[i] = replicasetConf
+		i++
+	}
+
+	if len(replicasetsList) == 0 {
```

Comment:
Seems it's the same as `len(replicasetsConf) == 1` and could be done above.

Reply:
You mean `len(replicasetsConf) == 0`? Yes, you are right.

Reply:
yes, `== 0`